app/models/batch_connect/app.rb

```diff
@@ -93,13 +99,19 @@ def cluster
     # Whether this is a valid app the user can use
     # @return [Boolean] whether valid app
     def valid?
+      return false if @validation_reason
```

Comment:
This won't be set until `#form_config` has been called. So something like `App.new(router: router).valid?` would skip this line, but it would still return `false` when it called `cluster` in the line below with the proper validation reason.

Although the proper error is still captured, it is confusing which line of code was responsible for it.

So one suggestion would be...

```ruby
def valid?
  !form_config.empty? && cluster && cluster.job_allow?
end
```

This makes it slightly clearer what we are testing for.